core/pkg/launch/job_builder.go

```diff
@@ -768,3 +777,35 @@ func (j *JobBuilder) HandleLogArtifactResult(response *service.LogArtifactRespon
 		}
 	}
 }
+
+// Configure a new job input for the job builder.
+//
+// This method is called when a user declares a new variable input for their
+// job. The request specifies the source of the input (a file or the run config)
+// and sets of keys in that config to include or exclude from the input. The
+// key sets are expressed as path prefixes in the config.
+func (j *JobBuilder) HandleJobInputRequest(request *service.JobInputRequest) {
+	// If job builder is disabled. This happens if run is created from a job.
+	if j == nil {
+		return
+	}
+	j.saveShapeToMetadata = true
+	j.logger.Debug("jobBuilder: handling job input request")
+	if request == nil {
+		j.logger.Error("jobBuilder: job input request is nil")
+		return
+	}
+	source := request.GetSource()
+	switch source.GetSource().(type) {
```

Comment:
```suggestion
	switch request.GetSource().GetSource().(type) {
```

Reply:
really confusing that there are 2 sources it seems like a typo unless you look at the proto

Reply:
Good call, I changed the top level field to input_source